### ISSUE TITLE
UIApplication:sharedApplication is used in YTPlayerView preventing it's use in App Extensions

### ISSUE THREAD

**Issue by jacksondeane** [youtube/youtube-ios-player-helper#193]
`YTPlayerView.m:657:21: sharedApplication is unavailable: not available on iOS (App Extension) - Use view controller based solutions where appropriate instead.`

Using `[[UIApplication sharedApplication] openURL:url];` prevents YTPlayerView from being used in code that is linked in an App Extension.

Maybe have `YTPlayerView` be initialized with an instance of `UIApplication`?

**Comment by JALsnipe** [comment]
Are you trying to use YTPlayerView in a application extension?  What exactly are you trying to do?

**Comment by jacksondeane** [comment]
Yes, our application let's the user preview and publish video and we are looking to support native YouTube playback.  We have a framework for this flow that is shared between an application and share extension.
